From 8c5608082d3fb845ffb1bd3e7b0ffeee2a1cafda Mon Sep 17 00:00:00 2001
From: Andreas Oberritter <obi@opendreambox.org>
Date: Fri, 15 Aug 2014 20:43:41 +0200
Subject: [PATCH 17/19] pkg_conflicts: Consider version constraints
To: opkg-devel@googlegroups.com
Cc: paul@paulbarker.me.uk

With packages A (Version: 2.0) and B (Conflicts: A (<< 1.0)) installed,
it wasn't possible to upgrade A to a newer version since commit 2e2329b.

Signed-off-by: Andreas Oberritter <obi@opendreambox.org>
---
 libopkg/pkg_depends.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libopkg/pkg_depends.c b/libopkg/pkg_depends.c
index ce48127..b23caf6 100644
--- a/libopkg/pkg_depends.c
+++ b/libopkg/pkg_depends.c
@@ -635,7 +635,8 @@ int pkg_conflicts(pkg_t *pkg, pkg_t *conflictee)
 	  for (j = 0; j < possibility_count; j++) {
 	       possibility = possibilities[j]->pkg;
 	       for (k = 0; k < conflictee_provides_count; k++) {
-		    if (possibility == conflictee_provides[k]) {
+		    if (possibility == conflictee_provides[k] &&
+                        version_constraints_satisfied(possibilities[j], conflictee)) {
 			 return 1;
 		    }
 	       }
-- 
1.9.1

